File: MoonEngine-Root/Base/src/main/java/com/greentree/engine/moon/base/layer/Layer.java
package com.greentree.engine.moon.base.layer;

import java.util.Objects;

import com.greentree.engine.moon.ecs.component.ConstComponent;

public record Layer(String value) implements ConstComponent {
	
	
	public Layer {
		Objects.requireNonNull(value);
	}
	
	@Override
	public String toString() {
		return "Layer [" + value + "]";
	}
	
}

File: MoonEngine/Base/src/main/java/com/greentree/engine/moon/base/layer/Layer.kt
package com.greentree.engine.moon.base.layer

import com.greentree.engine.moon.ecs.component.ConstComponent
import java.util.*

data class Layer(val name: String) : ConstComponent {

	fun name() = name

	override fun toString(): String {
		return "Layer [$name]"
	}

	init {
		Objects.requireNonNull(name)
	}
}